### PR TITLE
[GNA] Fuse gather and transpose

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Added transformation to replace the following patterns: Gather-Reshape-Transpose and Transpose-Reshape-Gather witj one Gather layer

### Tickets:
 - 115915

### CODE REVIEW

File: src/plugins/intel_gna/src/transformations/gather_sinking_transpose.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * @brief Merge Gather with Transpose through Reshape.
+ *
+ *      Any1[a,b,c]          Any1[a,b,c]
+ *       |                     |
+ *    Reshape[1,a*b*c]      Reshape[1,a*b*c]
+ *       |                     |
+ *     Gather[1,a*b*c]      Gather[1,a*b*c]
+ *       |                     |
+ *    Reshape[a,b,c]           |
```
Comment (sirzabek) [inline src/plugins/intel_gna/src/transformations/gather_sinking_transpose.hpp:23]: I think we should indicate here we're detecting patterns with possibly multiple reshapes at this position.

File: src/plugins/intel_gna/src/transformations/gather_sinking_transpose.hpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+class GatherSinkingTransposeForward : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("GatherSinkingTransposeForward", "0");
+    GatherSinkingTransposeForward();
+};
+
+/**
+ * @brief Merge Transpose with Gather through Reshape.
+ *
+ *      Any1[a,b,c]            Any1[a,b,c]
```
[the remark this answers is not in the document]
Comment (mryzhov) [reply src/plugins/intel_gna/src/transformations/gather_sinking_transpose.hpp:42]: No, this is just an example

File: src/plugins/intel_gna/src/transformations/gather_sinking_transpose.hpp (hunk 1 above)
Comment (mryzhov) [reply src/plugins/intel_gna/src/transformations/gather_sinking_transpose.hpp]: updated commnets

File: src/plugins/intel_gna/src/transformations/gather_sinking_transpose.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/gather_sinking_transpose.hpp"
+
+#include <utility>
+
+#include "openvino/cc/ngraph/itt.hpp"
+// #include "transformations/utils/utils.hpp"
```
[the remark this answers is not in the document]
Comment (mryzhov) [reply src/plugins/intel_gna/src/transformations/gather_sinking_transpose.cpp]: removed

**Review by sirzabek — APPROVED** [review]

**Pull request merged**